Firestore/core/test/unit/model/field_value_test.cc

```diff
@@ -1,776 +0,0 @@
-      // -0.0 and 0.0 compare the same but are not equal.
-      .AddEqualityGroup(Value(-0.0))
-      .AddEqualityGroup(Value(0.0))
-      .AddEqualityGroup(Value(1), FieldValue::FromInteger(1LL))
-      // Doubles and Longs aren't equal (even though they compare same).
-      .AddEqualityGroup(Value(1.0), FieldValue::FromDouble(1.0))
-      .AddEqualityGroup(Value(1.1), FieldValue::FromDouble(1.1))
-      .AddEqualityGroup(BlobValue(0, 1, 1))
-      .AddEqualityGroup(BlobValue(0, 1))
-      .AddEqualityGroup(Value("string"), FieldValue::FromString("string"))
-      .AddEqualityGroup(Value("strin"))
-      // latin small letter e + combining acute accent
-      .AddEqualityGroup(Value("e\u0301b"))
-      // latin small letter e with acute accent
-      .AddEqualityGroup(Value("\u00e9a"))
-      .AddEqualityGroup(Value(Timestamp::FromTimePoint(kDate1)),
-                        Value(kTimestamp1))
-      .AddEqualityGroup(Value(Timestamp::FromTimePoint(kDate2)),
-                        Value(kTimestamp2))
-      // NOTE: ServerTimestampValues can't be parsed via Value().
-      .AddEqualityGroup(FieldValue::FromServerTimestamp(kTimestamp1),
-                        FieldValue::FromServerTimestamp(kTimestamp1))
-      .AddEqualityGroup(FieldValue::FromServerTimestamp(kTimestamp2))
-      .AddEqualityGroup(Value(GeoPoint(0, 1)),
-                        FieldValue::FromGeoPoint(GeoPoint(0, 1)))
-      .AddEqualityGroup(Value(GeoPoint(1, 0)))
-      .AddEqualityGroup(FieldValue::FromReference(DbId(), Key("coll/doc1")),
-                        FieldValue::FromReference(DbId(), Key("coll/doc1")))
-      .AddEqualityGroup(
-          FieldValue::FromReference(DbId("project/bar"), Key("coll/doc2")))
-      .AddEqualityGroup(
-          FieldValue::FromReference(DbId("project/baz"), Key("coll/doc2")))
-      .AddEqualityGroup(Array("foo", "bar"), Array("foo", "bar"))
-      .AddEqualityGroup(Array("foo", "bar", "baz"))
-      .AddEqualityGroup(Array("foo"))
-      .AddEqualityGroup(WrapObject("bar", 1, "foo", 2),
-                        WrapObject("foo", 2, "bar", 1))
-      .AddEqualityGroup(WrapObject("bar", 2, "foo", 1))
-      .AddEqualityGroup(WrapObject("bar", 1))
-      .AddEqualityGroup(WrapObject("foo", 1))
-      .TestEquals();
-}
-
-#if __APPLE__
-// Validates that NSNumber/CFNumber normalize NaNs to the same values that
-// Firestore does. This uses CoreFoundation's CFNumber instead of NSNumber just
-// to keep the test in a single file.
-TEST_F(FieldValueTest, CanonicalBitsAreCanonical) {
-  double input = ToDouble(kAlternateNanBits);
-  CFNumberRef number = CFNumberCreate(nullptr, kCFNumberDoubleType, &input);
-  util::Defer cleanup([&] { util::SafeCFRelease(number); });
-
-  double actual = 0.0;
-  CFNumberGetValue(number, kCFNumberDoubleType, &actual);
-
-  ASSERT_EQ(kCanonicalNanBits, ToBits(actual));
-}
-#endif  // __APPLE__
-
-TEST_F(FieldValueTest, NormalizesNaNs) {
```

Comment:
This test only tested the test helpers since we remove the round trip through the FieldValues. It was removed.